src/map.ts

```diff
@@ -76,6 +87,31 @@ export namespace map {
   export function indexed() {
     return purry(_map(true), arguments, map.lazyIndexed);
   }
+
   export const lazy = _lazy(false);
   export const lazyIndexed = _toLazyIndexed(_lazy(true));
+
+  export function strict<T extends ReadonlyArray<unknown> | [], K>(
+    array: Readonly<T>,
+    fn: Pred<T[number], K>
+  ): Mapped<T, K>;
+  export function strict<T extends ReadonlyArray<unknown> | [], K>(
+    fn: Pred<T[number], K>
+  ): (array: Readonly<T>) => Mapped<T, K>;
+  export function strict(...args: ReadonlyArray<unknown>): unknown {
+    return purry(_map(false), args, map.lazy);
```

Comment:
instead of re-implementing `map` inside `strict`, could we just invoke the regular `map` function - similar to how `keys` is implemented? I realize now that `groupBy` also doesn't do that ...